azure/scope/managedcontrolplane.go

```diff
@@ -903,3 +905,35 @@ func (s *ManagedControlPlaneScope) PrivateEndpointSpecs() []azure.ASOResourceSpe
 func (s *ManagedControlPlaneScope) SetOIDCIssuerProfileStatus(oidc *infrav1.OIDCIssuerProfileStatus) {
 	s.ControlPlane.Status.OIDCIssuerProfile = oidc
 }
+
+// AKSExtension returns the cluster AKS extensions.
+func (s *ManagedControlPlaneScope) AKSExtension() []infrav1.MarketplaceExtension {
+	return s.ControlPlane.Spec.MarketplaceExtensions
+}
+
+// AKSExtensionSpecs returns the AKS extension specs.
+func (s *ManagedControlPlaneScope) AKSExtensionSpecs() []azure.ASOResourceSpecGetter[*asokubernetesconfigurationv1.Extension] {
+	if s.AKSExtension() == nil {
+		return nil
+	}
+	extensionSpecs := make([]azure.ASOResourceSpecGetter[*asokubernetesconfigurationv1.Extension], 0, len(s.ControlPlane.Spec.MarketplaceExtensions))
+	for _, extension := range s.ControlPlane.Spec.MarketplaceExtensions {
+		extensionSpec := &aksextensions.AKSExtensionSpec{
+			Name:                    extension.Name,
+			Namespace:               s.Cluster.Namespace,
+			AutoUpgradeMinorVersion: extension.AutoUpgradeMinorVersion,
+			ConfigurationSettings:   extension.ConfigurationSettings,
+			ExtensionType:           extension.ExtensionType,
+			ReleaseTrain:            extension.ReleaseTrain,
+			Version:                 extension.Version,
+			Owner:                   azure.ManagedClusterID(s.SubscriptionID(), s.ResourceGroup(), s.ControlPlane.Name),
+			OwnerRef:                *metav1.NewControllerRef(s.ControlPlane, infrav1.GroupVersion.WithKind(infrav1.AzureManagedControlPlaneKind)),
+			Plan:                    *extension.Plan,
+			// AKSAssignedIdentityType: extension.AKSAssignedIdentityType,
```

Comment:
Q: I know this is WIP but I'm curious if this is meant to be included.

Reply:
Yeah I just haven't added the field in the type yet, although this is now fully implemented in the latest commit.